.github/workflows/bicep_az_deployment_group.yml

```diff
@@ -0,0 +1,232 @@
+name: Perform action for an Az Deployment Group via Bicep
```

Comment:
Infuriatingly the `arm-deploy` action, doesn't like special characters in secrets so we write our own.

When reviewing, it might make sense to start with `bicep_workflow.yml`, then do `bicep_deploy.yml` then this file, as that's the hierarchy.